test/integration/operator_test.go

```diff
@@ -0,0 +1,227 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+//go:build integration
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/avast/retry-go/v4"
+	kubearchivev1alpha1 "github.com/kubearchive/kubearchive/cmd/operator/api/v1alpha1"
+	"github.com/kubearchive/kubearchive/test"
+	corev1 "k8s.io/api/core/v1"
+	errs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type Test int
+
+const (
+	emptyKAC Test = iota
+	nonEmptyKAC
```

Comment:
These are good tests, but they test only the "local" creation of kac resources, without any global kac in place.

IIUC, when a global KAC is in place, the "empty" local KAC is allowed, and I think that would be great to test.

WDYT?

Reply:
I will handle this as part of https://github.com/kubearchive/kubearchive/issues/863, which I just opened. I have a number of PRs currently working in this same area and want to get some resolved.

> IIUC, when a global KAC is in place, the "empty" local KAC is allowed, and I think that would be great to test.

Just to clarify, an empty `KubeArchiveConfig` is always possible.  A global `KubeArchiveConfig` is not required.

Reply:
yes my bad, I'll rephrase it: an empty `KubeArchiveConfig` is it's not created with a global one, it won't trigger the creation of an `APIServerSource` but with a global `KubeArchiveConfig` that it's not empty, the `APIServerSource` will be created, right?

Reply:
It's a little more general than that.  An `ApiServerSource` is only created when there are resources for it.  So the first `KubeArchiveConfig` that has resources will result in an `ApiServerSource` being created.  Conversely, when there are no `KubeArchiveConfigs` that have resources, the `ApiServerSource` will be removed.